examples/hosts/app-integration/external-data/src/mock-service-interface/dataParsing.ts

```diff
@@ -24,14 +30,13 @@ export interface ParsedTaskData {
  *
  * TODO: See notes below about moving away from plain string to something more realistic.
  */
-export function parseStringData(stringData: string): ParsedTaskData[] {
-    const taskStrings = stringData.split("\n");
-    return taskStrings.map((taskString) => {
-        const [taskIdString, taskNameString, taskPriorityString] = taskString.split(":");
+export function parseTaskData(taskData: object): ParsedTaskData[] {
+    const preparsed = taskData as TaskData;
+    return Object.entries(preparsed).map((task) => {
```

Comment:
Nit: This might be made a bit easier to read by doing something like:

```suggestion
    return Object.entries(preparsed).map(([taskId, taskBody]) => {
```

That way we don't have magic number indexes in the return body below.